### DISCUSSION TITLE
ServerGrabberDualDragonBayer640_480.ini gives black and white output only

### DISCUSSION THREAD

**Discussion by paliasgh** [robotology/community#616] [Q&A] [answered]
Hello,

We are trying to get 640X480 video output from iCub cameras and are running yarpdev module with "--from camera/ServerGrabberDualDragonBayer640_480.ini --split true". The resulting output has 640X480 resolution but it is black and white. How can we get RGB output?

Best,
Pourya

**Comment by pattacini** [comment]
Hi @paliasgh 

I converted your issue into a discussion and moved it here, as this place is meant for hosting this type of Q&A.
See https://github.com/robotology/community.

**Comment by martinaxgloria** [comment]
Hi @paliasgh,

did you try to run the cameras without the bayer? I mean running the yarpdev module with `--from camera/ServerGrabberDualDragon.ini --split true`, instead of camera/ServerGrabberDualDragonBayer640_480.ini

Let me know if the problem is still present or not. 
Bests,
Martina

**Reply by patrickshas** [reply]
Hi @martinaxgloria , yes it runs fine without the Bayer but it only gives a 320 by 240 output. We do need a higher resolution output (640 by 480) and hence trying to trouble shoot Bayer. Hope this explains further.

**Reply by martinaxgloria** [reply, marked as answer]
Hi @patrickshas 
A possibility is to change in `ServerGrabberDualDragonBayer640_480.ini` the parameter `capabilities` from `RAW` to `COLOR` and then in `camera/dragonfly2_config_left_bayer_640_480.ini` and `camera/dragonfly2_config_right_bayer_640_480.ini` the `device` from `dragonfly2raw` to `dragonfly2`.

**Reply by paliasgh** [reply]
Hi @martinaxgloria, thanks! Where exactly are those files located? We have a copy of them inside `robots-configuration/iCubWaterloo01/camera/` on our local computer. It seemed like modifying those files won't affect anything. Does `yarpdev` find the `.ini` files somewhere else? Maybe in `icub-head`?

**Reply by martinaxgloria** [reply]
Hi @paliasgh,
This is quite strange since you are in the right directory. Another attempt that it could be done is to make connection using the Bayer carrier. After running the yarpdev module, you can try to run `yarp connect /a /b mcast+recv.bayer+method.nearest`, where `/a` and `/b` are respectively your grabber port and the yarpview one.

**Reply by pattacini** [reply]
Just a couple of notes.

@martinaxgloria verified with @S-Dafarra and @lrapetti that updating the value of `capabilities` in the conf file plus changing the name of the device made the difference in their setup, as they were experiencing the same black-and-white output beforehand.

The conf files are those located in `icub-head` as per https://github.com/robotology/icub-main/blob/master/app/default/scripts/cameras_calib_bayer_640_480.xml.template#L6-L14 (where `icub-head` replaces `pc104`).

To known which config file gets actually loaded by the driver, you can use the following command:
```console
yarp resource --from camera/dragonfly2_config_right_bayer_640_480.ini
```
which employs the `YARP_ROBOT_NAME` env variable to browse within all the available contexts complying with priorities. Then, inspect the content of the file just found.

Find out more on apps and robots contexts as well as `yarp resource` in #42.

**Reply by patrickshas** [reply]
Many thanks @martinaxgloria , @pattacini , @S-Dafarra , @lrapetti . We'll implement this solution and will report back.

**Reply by patrickshas** [reply]
Hello all,
We were able to resolve the issue. Many thanks for all the suggestions.

We are now getting a 640 by 480 RGB image. 

We are facing a related but separate issue now. I'll cover this issue in the next reply in case it needs to be broken into a new discussion.
Thanks again.

**Reply by pattacini** [reply]
Happy you sorted it out!

> I'll cover this issue in the next reply

Better off opening up a brand-new discussion anyway.
You can always point it back to here for cross reference.
